pkg/networkservice/connectioncontext/ipcontext/routes/client.go

```diff
@@ -71,8 +71,10 @@ func (r *routesClient) Request(ctx context.Context, request *networkservice.Netw
 }
 
 func (r *routesClient) Close(ctx context.Context, conn *networkservice.Connection, opts ...grpc.CallOption) (*empty.Empty, error) {
-	if err := addDel(ctx, conn, r.vppConn, metadata.IsClient(r), false); err != nil {
+	_, err := next.Client(ctx).Close(ctx, conn, opts...)
+	if err != nil {
 		return nil, err
 	}
-	return next.Client(ctx).Close(ctx, conn, opts...)
+	err = addDel(ctx, conn, r.vppConn, metadata.IsClient(r), false)
+	return nil, err
```

Comment:
This produces a new issue in that if ```next.Client(ctx).Close(ctx,conn,opts...)``` returns an error, we never clean up the routes.

```suggestion
         _ = addDel(ctx, conn, r.vppConn, metadata.IsClient(r), false)
	return next.Client(ctx).Close(ctx, conn, opts...)
```